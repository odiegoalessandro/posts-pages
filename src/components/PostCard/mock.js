export const mock = {
  title: 'title',
  content: 'content',
  id: 1,
  cover: 'img/img.png'
};