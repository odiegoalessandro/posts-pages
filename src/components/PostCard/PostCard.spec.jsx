import { render, screen } from "@testing-library/react";
import { PostCard } from ".";
import { mock } from './mock';

describe('<PostCard />', () => {
  it('should render PostCard correctly', () => {
    render(<PostCard  {...mock} />);
  
    expect(screen.getByRole('img')).toHaveAttribute('src', mock.cover);
    expect(screen.getByRole('heading', { name: mock.title })).toBeInTheDocument();
    expect(screen.getByText(mock.content)).toBeInTheDocument();
  });

  it('should match with snapshot', () => {
    const { asFragment } = render(<PostCard {...mock} />);

    expect(asFragment()).toMatchSnapshot();
  });
});