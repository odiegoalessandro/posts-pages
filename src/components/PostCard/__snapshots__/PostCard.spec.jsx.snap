// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match with snapshot 1`] = `
<DocumentFragment>
  <div
    class="post"
  >
    <img
      alt="title"
      src="img/img.png"
    />
    <div
      class="post-content"
    >
      <h2>
        title
      </h2>
      <p>
        content
      </p>
    </div>
  </div>
</DocumentFragment>
`;
